frontend/src/main/components/Commons/CommonsTable.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+// import { toast } from "react-toastify";
+import { useBackendMutation } from "main/utils/useBackend";
+import { cellToAxiosParamsDelete, onDeleteSuccess } from "main/utils/commonsUtils"
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function CommonsTable({ commons, currentUser }) {
+
+    const navigate = useNavigate();
+
+    const editCallback = (cell) => {
+        navigate(`/commons/edit/${cell.row.values.id}`)
+    }
+
+    // Stryker disable all : hard to test for query caching
+
+    const deleteMutation = useBackendMutation(
+        cellToAxiosParamsDelete,
+        { onSuccess: onDeleteSuccess },
+        ["/api/commons/all"]
+    );
+    // Stryker enable all 
+
+    // Stryker disable next-line all : TODO try to make a good test for this
+    const deleteCallback = async (cell) => { deleteMutation.mutate(cell); }
+
+
+    const columns = [
+        {
+            Header: 'id',
+            accessor: 'id', // accessor is the "key" in the data
+            
+        },
+        {
+            Header:'name',
+            accessor: 'name',
+        },
+        {
+            Header:'day',
+            accesor:'day',
+        },
+        {
+            Header:'startDate',
+            accesor:'startDate',
+        },
+        {
+            Header:'endDate',
+            accesor:'endDate',
+        },
+        {
+            Header:'totalPlayers',
+            accesor:'totalPlayers',
+        },
+        {
+            Header:'cowPrice',
+            accesor:'cowPrice',
+        },
+        {
+            Header:'milkPrice',
+            accesor:'milkPrice',
+        },
+    ];
+
+    if (hasRole(currentUser, "ROLE_ADMIN")) {
+        columns.push(ButtonColumn("Edit", "primary", editCallback, "CommonsTable"));
+        columns.push(ButtonColumn("Delete", "danger", deleteCallback, "CommonsTable"));
+    } 
+
+    // Stryker disable next-line ArrayDeclaration : [columns] is a performance optimization
+    const memoizedColumns = React.useMemo(() => columns, [columns]);
+    const memoizedCommons = React.useMemo(() => commons, [commons]);
```

Comment:
This does not actually memoize these fields. In order for this to actually be a performance optimization, the function within `useMemo` must generate the array. As it is currently, this actually negatively impacts performance.